aten/src/ATen/native/BucketizationUtils.h

```diff
@@ -134,6 +134,12 @@ inline void searchsorted_pre_check(
 
     TORCH_CHECK(sorter.scalar_type() == ScalarType::Long, "torch.searchsorted(): sorter must be a tensor of long ",
       "dtype but got dtype ", sorter.scalar_type());
+
+    if (sorter.numel() > 0) {
+      const long innermost_size = sorter.sizes().back();  // get inner most dimension of sorter
+      TORCH_CHECK(sorter.max().item().toLong() < innermost_size, "torch.searchsorted(): sorter index out of range");
+      TORCH_CHECK(sorter.min().item().toLong() >= 0, "torch.searchsorted(): sorter index out of range");
```

Comment:
Documentation is a bit obscure here, but aren't indices in Pytorch usually wrap around (i.e. `min()` should be `>= -inntermost_size`)

Reply:
I agree. But the `searchsorted` implementation seems did not well consider such cases.

![image](https://user-images.githubusercontent.com/38074777/218938200-69941959-c641-4bfe-baff-1932723c27ce.png)

`0` and `-1` in sorter should represent the same indices for a single-elem vector but the results are different. Not sure if we wanted to fix it and allow such usage, or just disable it for now.

Reply:
nit: use `aminmax` for slightly better perf.